pkg/controller/storagecluster/cephcluster.go

```diff
@@ -550,6 +550,14 @@ func (r *ReconcileStorageCluster) checkTuneStorageDevices(ds ocsv1.StorageDevice
 		return dt.speed, nil
 	}
 
+	ret, err := r.TunePlatformForFastDevices()
```

Comment:
Probably a more simpler implementation would be to check if `TuneFastPlatforms` contains `r.platform.GetPlatform()`.
That way I wouldn't have to jump to definition of `TunePlatformForFastDevices` to understand what is happening and why we are tuning.

Currently it seems like `TunePlatformForFastDevices` is doing tuning intensive stuff while in reality it's just checking if `TuneFastPlatforms` contains `r.platform.GetPlatform()`.

As a follow-up I'd look into re-factoring platform detection and maybe storing it in ReconcileStorageCluster if it makes more sense.

Reply:
@obnoxxx ^^

Reply:
For me I feel both ways are good and maybe if adding comment solve this problem that is also a way to go.

Reply:
"If your code needs a comment to explain what it does, it isn't really a readable/understandable code". 
Read this somewhere and I relate.

We could inline the call, rename the method, use better variables etc to make it better. Adding a comment would be the last thing I'd suggest.

Reply:
> Probably a more simpler implementation would be to check if `TuneFastPlatforms` contains `r.platform.GetPlatform()`.

Well, that is exactly what the function does!?

> That way I wouldn't have to jump to definition of `TunePlatformForFastDevices` to understand what is happening and why we are tuning.

That's more a naming thing than an implementation thing. I explicitly asked @crombus to write such a function to *not* have the implementation bleed into the caller. I didn't want to repeat the (imho) bad approach of `aviodObjectStore()` where each caller first has to go and call `GetPlatform` and then insert it into `avoidObjectStore`. This is bad encapsulation/abstraction, and error prone. If there is more than one caller, it's also code duplication.

So it boils down to: we only need to name the function / vars better to make it more obvious what is going on.

> Currently it seems like `TunePlatformForFastDevices` is doing tuning intensive stuff while in reality it's just checking if `TuneFastPlatforms` contains `r.platform.GetPlatform()`.

Right, there needs to be a better name. I like it when in some languages, like ruby, functions can end with a '?' symbol. That makes it more obvious. But here we don't have that option. We could do something like this:

```
tuneFastDevices, err := r.DevicesDefaultToFastForThisPlatform()
...
```

or 

```
tuneDevicesFast, err := DoTuneDevicesFastForPlatform()
...
```
or similar...

> As a follow-up I'd look into re-factoring platform detection and maybe storing it in ReconcileStorageCluster if it makes more sense.

Possible. I have started a discussion with @crombus already about possible refactorings of the platform-detection code triggered by this change. Storing data sounds like an optimization. But let's get the APIs good in the first place. And let's not add new stuff that needs immediate refactoring afterwards. And to reiterate my initial point: I think the current approach is the right encapsulation and just needs better naming.

Reply:
@crombus - can you rename the function and variable along the above lines to make the code more obvious?